src/goo/fsmpack/statemachine/StateMachineSystem.js

```diff
@@ -62,27 +61,11 @@ define([
 	};
 
 	StateMachineSystem.prototype.process = function (entities, tpf) {
-		var component;
-
-		if (this.resetRequest) {
-			this.resetRequest = false;
-			for (var i = 0; i < entities.length; i++) {
-				component = entities[i].stateMachineComponent;
-				component.kill();
-				component.cleanup();
-			}
-			this.time = 0;
-			// remove all sounds a bit hard but in reality no tween should remain alive between runs
-			TWEEN.removeAll();
-			this.passive = true;
-			return;
-		}
-
 		this.time += tpf;
 
 		// Enter unentered components
 		for (var i = 0; i < entities.length; i++) {
-			component = entities[i].stateMachineComponent;
+			var component = entities[i].stateMachineComponent;
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: ["i" used outside of binding context.](https://www.codacy.com/app/rikard/goojs/file/2319176656/issues/source?bid=3193730&fileBranchId=3204236#l68)